tests/MP3.test.ts

```diff
@@ -0,0 +1,86 @@
+import { strict as assert } from 'node:assert';
+import * as fs from 'node:fs';
+import * as path from 'node:path';
+import { MP3 } from '../src/asset';
+
+const fixturesPath = 'tests/fixtures';
+const getFixturePath = (fixture: string) => path.join(fixturesPath, fixture);
+
+async function createAssetWithManifest(manifestData: Uint8Array): Promise<MP3> {
+    const mp3Buffer = fs.readFileSync(getFixturePath('sample1.mp3'));
+    const asset = new MP3(mp3Buffer);
+    await asset.ensureManifestSpace(manifestData.length);
+    await asset.writeManifestJUMBF(manifestData);
+    return asset;
+}
+
+async function verifyManifestInNewInstance(asset: MP3, expectedManifest: Uint8Array | undefined): Promise<void> {
+    const modifiedBuffer = await asset.getDataRange();
+    const newAsset = new MP3(modifiedBuffer);
+    const newRetrievedManifest = newAsset.getManifestJUMBF();
+    assert.deepEqual(newRetrievedManifest, expectedManifest, 'manifest should match in new asset instance');
+}
+
+describe('MP3', function () {
```

Comment:
Functional tests that do full signing and validation of an MP3 file would be great.